### DISCUSSION TITLE
Potential Fix/Idea for Network Manager to find DefaultPrefabObjects in project if it is null!

### DISCUSSION THREAD

**Discussion by Katerlad** [FirstGearGames/FishNet#539] [Feature Request]
We could utilize the built in Resources Library in Unity to search for the DefaultPrefabObjects, if the NetworkManager has a null reference to it.

![image](https://github.com/FirstGearGames/FishNet/assets/54480322/2b3a4264-31f6-45ce-8d5a-778a5026aa58)

Changing the FishNet folder to have a "Resources" folder in it will allow you to search for "DefaultPrefabsObject" inside it without a taxing load. 

The Resource Library is not recommended for full projects with many assets, by Unity, but should suffice for a addon package like Fishnet to lookup one single asset.

You would have to make a folder inside FishNet called "Resources" and a folder inside that called "FishNetAssets" and inside there should be where the "DefaultPrefabObject" Scriptable Object lives. 

FishNet => Resources => FishNetAssets => DefaultPrefabObjects

Then the NetworkManager on Serialization into a scene  if the PrefabObjects is null! 
Do a Resources.Load< ScriptableObject >(FishNetAssets/DefaultPrefabObjects);

**Comment by FirstGearGames** [comment]
I believe we can actually just do this without the need for all that trouble. We can do a check on the NetworkManager outside play 'if null' then search the default save path, or whichever path devs specified if they configured it. Should be extremely cheap and fast.

**Comment by FirstGearGames** [comment]
Turns out I already had code I could use for this. Managed to implement it with a few lines of code.
Added in 4.2.3.

https://github.com/FirstGearGames/FishNet/assets/27169134/0ed1febf-7a2e-4a58-bfdf-bd4430b8e534
